### PR TITLE
Update TestData project to utilise persistence and domain projects

### PR DESCRIPTION
**Pull request by PhysicsJon**

### CODE REVIEW

File: src/OrderFormAcceptanceTests.TestData/Helpers/OrderingPartyHelper.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            if (existing is not null)
+            {
+                return orderingParty;
+            }
+
+            var query = @"INSERT INTO OrderingParty
+                            (
+                              Id,
+                              OdsCode,
+                              [Name]
```
Comment (bjss-chris) [inline src/OrderFormAcceptanceTests.TestData/Helpers/OrderingPartyHelper.cs:32]: Looks like you're missing a comma here.

File: src/OrderFormAcceptanceTests.TestData/Helpers/OrderingPartyHelper.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                              AddressId
+                            )
+                          VALUES (
+                              @Id,
+                              @OdsCode,
+                              @Name
```
Comment (bjss-chris) [inline src/OrderFormAcceptanceTests.TestData/Helpers/OrderingPartyHelper.cs:38]: And here (comma).

File: src/OrderFormAcceptanceTests.TestData/Models/CreateOrderItemModel.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+
+namespace OrderFormAcceptanceTests.TestData.Models
+{
+    class CreateOrderItemModel
```
Comment (bjss-chris) [inline src/OrderFormAcceptanceTests.TestData/Models/CreateOrderItemModel.cs:5]: `public sealed`?

File: src/OrderFormAcceptanceTests.TestData/Models/ItemUnitModel.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿namespace OrderFormAcceptanceTests.TestData.Models
+{
+    public class ItemUnitModel
```
Comment (bjss-chris) [inline src/OrderFormAcceptanceTests.TestData/Models/ItemUnitModel.cs:3]: `sealed`?

File: src/OrderFormAcceptanceTests.TestData/Models/OrderItemRecipientModel.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿namespace OrderFormAcceptanceTests.TestData.Models
+{
+    using System;
+
+    public class OrderItemRecipientModel
```
Comment (bjss-chris) [inline src/OrderFormAcceptanceTests.TestData/Models/OrderItemRecipientModel.cs:5]: `sealed`?

File: src/OrderFormAcceptanceTests.TestData/Models/TimeUnitModel.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿namespace OrderFormAcceptanceTests.TestData.Models
+{
+    using OrderFormAcceptanceTests.Domain;
+
+    public class TimeUnitModel
```
Comment (bjss-chris) [inline src/OrderFormAcceptanceTests.TestData/Models/TimeUnitModel.cs:5]: `sealed`?

File: src/OrderFormAcceptanceTests.TestData/OrderItemList.cs [hunk 7]
```diff
@@ -3,12 +3,13 @@
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using OrderFormAcceptanceTests.Domain;
 
-    public sealed class OrderItemList
+    public sealed class List
```
Comment (bjss-chris) [inline src/OrderFormAcceptanceTests.TestData/OrderItemList.cs:8]: Do you definitely want to rename this class to `List`? It still seems to be specific to order items and `List` is a bit vague. If you definitely want to rename it to `List` then recommend changing the file name to match too.

File: src/OrderFormAcceptanceTests.TestData/SupplierInfo.cs [hunk 8]
```diff
@@ -59,48 +29,43 @@ FROM CatalogueItem
...
 
-            return SqlExecutor.Execute<string>(connectionString, query, new { supplierId, itemType });
+            if (result is null)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
```
Comment (bjss-chris) [inline src/OrderFormAcceptanceTests.TestData/SupplierInfo.cs:63]: Can we simplify this to `return result is not null`?

File: src/OrderFormAcceptanceTests.TestData/SupplierInfo.cs [hunk 9]
```diff
@@ -59,48 +29,43 @@ FROM CatalogueItem
...
         }
 
-        private static IEnumerable<SupplierDetails> SupplierLookup(string connectionString, CatalogueItemType catalogueItemType, ProvisioningType provisioningType)
+        private static IEnumerable<SupplierDetails> SupplierLookup(string connectionString, CatalogueItemType catalogueItemType, Domain.ProvisioningType provisioningType)
         {
-            var query = @"SELECT ci.[SupplierId], su.[Name]      
+            var query = @"SELECT ci.[SupplierId], su.[Name], su.Address    
                             FROM [dbo].[CatalogueItem] ci
                             INNER JOIN CataloguePrice pr ON ci.CatalogueItemId=pr.CatalogueItemId
                             INNER JOIN Supplier su On ci.SupplierId=su.Id
```
Comment (bjss-chris) [inline src/OrderFormAcceptanceTests.TestData/SupplierInfo.cs:71]: Recommend including `AS` for table aliases, for example:

```sql
FROM dbo.CatalogueItem AS ci
```

Would recommend removing square brackets (except for `Name`) and adding spaces around operators. However, these are just non-essential suggestions that I think would improve readability – can always be done later as tech debt.

**Review by bjss-chris — CHANGES_REQUESTED** [review]

File: src/OrderFormAcceptanceTests.TestData/Models/CreateOrderItemModel.cs (hunk 3 above)
Comment (PhysicsJon) [reply src/OrderFormAcceptanceTests.TestData/Models/CreateOrderItemModel.cs]: I thought I had removed this file prior to pushing up, removed in latest commit

File: src/OrderFormAcceptanceTests.TestData/Helpers/OrderingPartyHelper.cs (hunk 2 above)
Comment (PhysicsJon) [reply src/OrderFormAcceptanceTests.TestData/Helpers/OrderingPartyHelper.cs]: I thought I had removed this file prior to pushing up, removed in latest commit

File: src/OrderFormAcceptanceTests.TestData/Helpers/OrderingPartyHelper.cs (hunk 1 above)
Comment (PhysicsJon) [reply src/OrderFormAcceptanceTests.TestData/Helpers/OrderingPartyHelper.cs]: I thought I had removed this method prior to pushing up, removed in latest commit

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**